src/compiler/scanner.ts

```diff
@@ -388,6 +416,22 @@ export function stringToToken(s: string): SyntaxKind | undefined {
     return textToToken.get(s);
 }
 
+const regExpFlagChars = makeReverseMap(charToRegExpFlag);
+
+/** @internal */
+export function regularExpressionFlagToCharacter(f: RegularExpressionFlags): string | undefined {
+    return regExpFlagChars[f];
+}
+
+/** @internal */
+export function characterToRegularExpressionFlag(c: string): RegularExpressionFlags | undefined {
+    return charToRegExpFlag.get(c);
+}
+
+function regularExpressionFlagToFirstAvailableLanguageVersion(f: RegularExpressionFlags): ScriptTarget | undefined {

```

Comment:
I would just inline usages of this function.

Reply:
This pattern isn't actually new; see the token code above.

Reply:
That one is exported and used pretty frequently, right?

I haven't seen where `regularExpressionFlagToCharacter` and `characterToRegularExpressionFlag` are used yet, but at least this one is only used in this file.